bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -265,6 +277,36 @@ object BagsManager {
   // This is our mutable state.  It is only read/written by the Swing thread.
   private var stateMap  = ==>>.empty[ProgKey, ObsKey ==>> BagsState]
 
+  def statusLookup(p: SPNodeKey, o: SPNodeKey): Option[BagsStatus] =
+    for {
+      obsMap <- stateMap.lookup(ProgKey(p))
+      s      <- obsMap.lookup(ObsKey(o))
+      st     <- s.status
+    } yield st
+
+  // Listeners for changes to the Bags Status.
+  // Management must be done on the Swing EDT.
+  private var listeners: List[BagsStatusListener] = Nil
+
+  def addBagsStatusListener(l: BagsStatusListener): Unit = {
```

Comment:
I think that is probably okay but the semantics are a bit different.  `Swing.onEDT` will add an event to the queue and return immediately so the listener isn't added yet at the time the method returns.  I don't know whether that matters here though.  We could at least break out the code for checking whether we're on the EDT so it isn't repeated.

````scala
private def assertingSwingEDT[A](message: String)(block: => A): A = 
  if (SwingUtilities.isEventDispatchThread) block
  else throw new IllegalStateException(message)
````

then calling it like

````scala
def addBagsStatusListener(l: BagsStatusListener): Unit =
  assertingSwingEDT("BagsStatusListeners can only be added on the Swing EDT.") {
    if (!listeners.contains(l)) listeners = l :: listeners
  }
````

or something.

Also, the `statusLookup` (which will become `stateLookup` I suppose) should also guarantee that it is being done on the EDT.